boards/common/openocd-nrf5.board.cmake

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Infer nrf51 vs nrf52 etc from the BOARD name. This enforces a board
+# naming convention: "nrf5x" must appear somewhere in the board name
+# for this to work.
+#
+# Boards which don't meet this convention can set this variable before
+# including this script.
+if (NOT DEFINED OPENOCD_NRF5_SUBFAMILY)
+  string(REGEX MATCH nrf5. OPENOCD_NRF5_SUBFAMILY "${BOARD}")
+endif()
+if("${OPENOCD_NRF5_SUBFAMILY}" STREQUAL "")
+  message(FATAL_ERROR
+    "Can't match nrf5 subfamily from BOARD name. "
+    "To fix, set CMake variable OPENOCD_NRF5_SUBFAMILY.")
+endif()
+
+# We can do the right thing for supported subfamilies using a generic
+# script, at least for openocd 0.10.0 and the openocd shipped with
+# Zephyr SDK 0.10.3.
+set(pre_init_cmds
+  "set WORKAREASIZE 0x4000"	# 16 kB RAM used for flashing
+  "source [find interface/jlink.cfg]"
+  "transport select swd"
+  "source [find target/${OPENOCD_NRF5_SUBFAMILY}.cfg]"
```

Comment:
@ioannisg there needs to be target/nrf53.cfg in the openocd distribution's scripts directory, or you need to:

1. make sure it works with e.g. nrf52.cfg
2. manually set OPENOCD_NRF5_SUBFAMILY to nrf52 in the nrf53 board.cmake
3. include this script (as done for other boards)

Either way, I don't think we need to gate merging this PR on nRF53 support getting merged. This PR works now for a large number of already-released devices.

Reply:
Oh, man, yes, that's why you didn't make this for nrf91 as well :p

Reply:
right :)